search-handlers/src/test/java/no/unit/nva/search/SearchResourceAuthHandlerTest.java

```diff
@@ -86,20 +90,99 @@ void shouldReturnOkWhenUserIsEditor() throws IOException {
     prepareRestHighLevelClientOkResponse();
 
     var input =
-        getInputStreamWithAccessRight(randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL);
+        getInputStreamWithAccessRight(
+            randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL, V_LEGACY);
+    handler.handleRequest(input, outputStream, contextMock);
+
+    var gatewayResponse = FakeGatewayResponse.of(outputStream);
+
+    assertNotNull(gatewayResponse.headers());
+    assertEquals(HTTP_OK, gatewayResponse.statusCode());
+  }
+
+  @Test
+  void shouldDefaultToLegacyModelWhenNoVersionSpecified() throws IOException {
+    prepareRestHighLevelClientOkResponse();
+
+    var input =
+        getInputStreamWithAccessRight(
+            randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL, null);
+    handler.handleRequest(input, outputStream, contextMock);
+
+    var gatewayResponse = FakeGatewayResponse.of(outputStream);
+
+    assertNotNull(gatewayResponse.headers());
+    assertEquals(HTTP_OK, gatewayResponse.statusCode());
+    assertNotNull(gatewayResponse.body());
+    assertThat(gatewayResponse.body().hits().size(), is(equalTo(2)));
+
+    var firstHitJson = gatewayResponse.body().hits().getFirst();
+    assertLegacyModel(firstHitJson);
+  }
+
+  @Test
+  void shouldReturnSimplifiedModelWhenVersion20241201Specified() throws IOException {
+    prepareRestHighLevelClientOkResponse();
+
+    var input =
+        getInputStreamWithAccessRight(
+            randomUri(), randomUri(), AccessRight.MANAGE_RESOURCES_ALL, V_2024_12_01_SIMPLER_MODEL);
     handler.handleRequest(input, outputStream, contextMock);
 
     var gatewayResponse = FakeGatewayResponse.of(outputStream);
 
     assertNotNull(gatewayResponse.headers());
     assertEquals(HTTP_OK, gatewayResponse.statusCode());
+
+    var firstHitJson = gatewayResponse.body().hits().getFirst();
+    var firstHitDto =
+        objectMapperWithEmpty.treeToValue(
+            firstHitJson, no.unit.nva.search.resource.response.ResourceSearchResponse.class);
+
+    assertNotNull(firstHitDto);
+    assertNotNull(firstHitDto.id());
+
+    assertSimplifiedModel(firstHitJson);
```

Comment:
Same question here really: Is it necessary to assert all these things in the same test? For example, `assertNotNull(firstHitDto.id());` would fail if `firstHitDto` is `null`.